tests/integration_tests/test_login.py

```diff
@@ -221,3 +224,213 @@ async def test_server_double_login(lobby_server):
         "style": "error",
         "text": "You have been signed out because you signed in elsewhere."
     }
+
+
+async def test_server_valid_login_with_token(lobby_server, api_priv_key):
+    proto = await connect_client(lobby_server)
+    await proto.send_message({
+        "command": "auth",
+        "version": "1.0.0-dev",
+        "user_agent": "faf-client",
+        "token": jwt.encode({
+          "user_id": 3,
+          "user_name": "Rhiza",
+          "scope": [],
+          "exp": int(time() + 10),
+          "authorities": [],
+          "non_locked": True,
+          "jti": "",
+          "client_id": ""
+        }, api_priv_key, algorithm="RS256").decode(),
+        "unique_id": "some_id"
+    })
+
+    msg = await proto.read_message()
+    assert msg["command"] == "irc_password"
+    msg = await proto.read_message()
+    me = {
+        "id": 3,
+        "login": "Rhiza",
+        "clan": "123",
+        "country": "",
+        "ratings": {
+            "global": {
+                "rating": [1650.0, 62.52],
+                "number_of_games": 2
+            },
+            "ladder_1v1": {
+                "rating": [1650.0, 62.52],
+                "number_of_games": 2
+            }
+        },
+        "global_rating": [1650.0, 62.52],
+        "ladder_rating": [1650.0, 62.52],
+        "number_of_games": 2
+    }
+    assert msg == {
+        "command": "welcome",
+        "me": me,
+        "id": 3,
+        "login": "Rhiza"
+    }
+    msg = await proto.read_message()
+    assert msg == {
+        "command": "player_info",
+        "players": [me]
+    }
+    msg = await proto.read_message()
+    assert msg == {
+        "command": "social",
+        "autojoin": ["#123_clan"],
+        "channels": ["#123_clan"],
+        "friends": [],
+        "foes": [],
+        "power": 0
+    }
+
+
+async def test_server_login_expired_token(lobby_server, api_priv_key):
+    proto = await connect_client(lobby_server)
+    await proto.send_message({
+        'command': 'auth',
```

Comment:
Oof  so many single quotes. Guess the base PR must be pretty dang old

Reply:
Done